financial-connections/src/main/java/com/stripe/android/financialconnections/domain/GoNext.kt

```diff
@@ -1,37 +1,69 @@
 package com.stripe.android.financialconnections.domain
 
+import com.stripe.android.core.Logger
 import com.stripe.android.financialconnections.model.FinancialConnectionsSessionManifest
+import com.stripe.android.financialconnections.model.FinancialConnectionsSessionManifest.FinancialConnectionsAuthorizationSession
 import com.stripe.android.financialconnections.model.FinancialConnectionsSessionManifest.NextPane
+import com.stripe.android.financialconnections.navigation.NavigationCommand
 import com.stripe.android.financialconnections.navigation.NavigationDirections
 import com.stripe.android.financialconnections.navigation.NavigationManager
 import javax.inject.Inject
 
+/**
+ * Usecase that determines the next step to navigate based on the received input.
+ *
+ * @see [FinancialConnectionsSessionManifest.nextPane]
+ * @see [FinancialConnectionsAuthorizationSession.nextPane]
+ *
+ */
 internal class GoNext @Inject constructor(
-    private val navigationManager: NavigationManager
+    private val navigationManager: NavigationManager,
+    private val logger: Logger
 ) {
 
-    operator fun invoke(manifest: FinancialConnectionsSessionManifest) {
-        navigationManager.navigate(
-            manifest.nextPane.toNavigationCommand()
-        )
+    operator fun invoke(
+        currentPane: NavigationCommand,
+        manifest: FinancialConnectionsSessionManifest,
+        authorizationSession: FinancialConnectionsAuthorizationSession?
+    ): NavigationCommand {
+        val nextPane = when (currentPane.destination) {
+            /**
+             * institution picker step receives a fresh [FinancialConnectionsAuthorizationSession]
+             * after picking a bank, source of truth for navigation.
+             */
+            NavigationDirections.institutionPicker.destination ->
+                authorizationSession!!.nextPane.toNavigationCommand()
```

Comment:
is this safe?

Reply:
Yes! institution picker always relies on authorizationSession to navigate to the next pane. 

authorizationSession is fetched when selecting an institution, right before calling `GoNext`, so it can't be null.